psi4/src/psi4/libmints/extern.h

```diff
@@ -61,6 +61,8 @@ class PSI_API ExternalPotential {
     std::vector<std::tuple<double, double, double, double> > charges_;
     /// Auxiliary basis sets (with accompanying molecules and coefs) of diffuse charges
     std::vector<std::pair<std::shared_ptr<BasisSet>, SharedVector> > bases_;
+    /// Gradient, if available, as number of charges x 3 SharedMatrix
+    SharedMatrix gradient_;
```

Comment:
I think member data named `gradient_` that *isn't* natom x 3 and `gradient_ != grad` in the .cc file is going to be confusing. I'd rename it `gradient_on_charges_` or something.

Reply:
I'm going to agree with Lori on this one. Building on her idea, maybe it's also worth naming the other gradient `grad` to `gradient_on_atoms` or something likewise, to clearly mark what each gradient in this part of the code refers to.

Reply:
Note that the `grad` isn't member data, just a local var, so it's name doesn't have to be so formal. I haven't checked but probably it gets stored on wfn in an array of natom x 3 grads. It's only the ncharge x 3 that's stored (newly) on the ExPot object. With the negative sign, should it be labeled a force?

Reply:
Let's stick with the gradient terminology over force, since the negative sign in the nuclear contribution comes from reusing the displacement used for the nuclear contribution on the `gradient_on_atoms` (it's a case of r_ij = -r_ji).